lib/runner.js

```diff
@@ -146,17 +147,22 @@ class Runner extends EventEmitter {
 	}
 
 	buildStats() {
+		this.tests.finalize();
```

Comment:
How does this work given that `TestCollection#finalize()` does the finalizing via `setImmediate()`?

I think it'd be better to get an array of pending tests from `TestCollection` and then iterate through it, calling `addTestResult()`. I'd then modify `addTestResult()` to take a second `pending` argument and provide the correct value here and in `run()` below. That way `Test` doesn't have to be aware of `pending` at all.

`TestCollection#finalize()` can then be removed. For clarity I think `buildStats()` should be renamed to `exit()`.

Reply:
Agreed - though this does burrow a bit into the `TestCollection`, I feel like adding such a loop at the top of `buildStats()` (or `exit()`, if we were to rename it) is a cleaner way to do this.

Reply:
> I feel like adding such a loop at the top of buildStats() (or exit(), if we were to rename it) is a cleaner way to do this.

Which is what you've done, right?